Update to operator-sdk v1.8.0

sjug opened the pull request:
In addition to the update, persist the correct functions for deployment etc inside of the `Makefile`.
This is the prerequisite step for the OLM bundle generation.

sjug commented:
Not sure what to do here?

```
From https://github.com/Mellanox/network-operator
 * [new ref]         refs/pull/206/head  -> origin/pull-requests/206/head
 * [new ref]         refs/pull/206/merge -> origin/pull-requests/206/merge
+ git checkout pull-requests/206/head
error: Your local changes to the following files would be overwritten by checkout:
	Makefile
Please commit your changes or stash them before you switch branches.
Aborting
+ let status=0+1
+ [[ 1 != \0 ]]
+ echo 'ERROR: Failed to checkout the nic-operator pull request number 206!'
ERROR: Failed to checkout the nic-operator pull request number 206!
```

AbdYsn commented:
@sjug sorry for that, 
already proposed a patch to fix it in CI 
see https://github.com/Mellanox/kubernetes-ci/pull/138

sjug commented on Makefile:
```diff
@@ -1,85 +1,42 @@
-DOCKERARGS=
-ifdef HTTP_PROXY
-	DOCKERARGS += --build-arg http_proxy=$(HTTP_PROXY)
+# VERSION defines the project version for the bundle.
```
Hey @adrianchiris,

Indeed, the Makefile has been auto-generated for many releases now. 
I'll try to bring some of that back. Not sure why one would specifically want to build with docker.

adrianchiris replied:
OK from the list above here is what is missing (*lint):

hadolint
helm-lint
lint (golangci) 

these are covered by github actions you addded. however to run them locally, one would need to install and run each command instead of invoking the Makefile target as it was before.

can you elaborate on why you chose to omit those ?

adrianchiris commented on Dockerfile:
```diff
@@ -1,42 +1,29 @@
+FROM gcr.io/distroless/static:nonroot
+WORKDIR /
+COPY --from=builder /workspace/manager .
+COPY manifests/ manifests/
```
original destination for `manifests` was `/etc/manifests` will this work ?
https://github.com/Mellanox/network-operator/blob/e73e49b99ff9d2f0eb92ab95143a9b4890f96d07/deploy/operator.yaml#L44

sjug replied:
Yep, works

adrianchiris commented on Makefile:
```diff
@@ -248,9 +220,58 @@ rm -rf $$TMP_DIR ;\
+
+.PHONY: bundle-build
+bundle-build: ## Build the bundle image.
+	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
```
should `${IMAGE_BUILDER}` be used here as well ?

sjug replied:
Yep, changed.

adrianchiris commented on Makefile:
```diff
@@ -71,65 +73,76 @@ HELM_VER = v3.5.3
-#endif
-#BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+# Docker vars
+IMAGE_BUILDER ?= podman
```
keep default as before (docker) ? we can change it later if needed but it will require changes in our CI

sjug replied:
Changed to `docker`

AbdYsn commented:
/retest-all

e0ne commented on Dockerfile:
```diff
@@ -1,42 +1,29 @@
-
-ENTRYPOINT ["/usr/local/bin/entrypoint"]
-USER ${USER_UID}
+# Build the manager binary
```
Please, leave license header

e0ne commented on config/default/manager_auth_proxy_patch.yaml:
```diff
@@ -10,6 +10,7 @@ spec:
     spec:
       containers:
       - name: kube-rbac-proxy
+        #image: gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
```
Why do we need this change?

sjug replied:
There's a bug with v0.8.0

e0ne commented on config/manager/manager.yaml:
```diff
@@ -23,7 +23,7 @@ spec:
     spec:
       securityContext:
-        runAsUser: 65532
+        runAsNonRoot: true
```
Why do we need root here?

sjug replied:
`runAsNonRoot` = not root

e0ne replied:
My bad, thanks

e0ne reviewed (changes requested):
Helm installation not tested yet but everything else seems to be working

e0ne commented:
'make lint' fails in several files

e0ne reviewed (changes requested):
need to change 'command' param in network-operator/templates/operator.yaml

This PR doesn't work for me if I install operator using helm  even I fixed template:
`2021-07-08T13:21:37.062Z        ERROR   controllers.NicClusterPolicy    Error creating state manager.   {"error:": "failed to create state manager: failed to create OFED driver State: failed to get files from manifest dir: error traversing directory tree: lstat /etc/manifests/stage-ofed-driver: no such file or directory", "error:Verbose": "lstat /etc/manifests/stage-ofed-driver: no such file or directory\nerror traversing directory tree\ngithub.com/Mellanox/network-operator/pkg/utils.GetFilesWithSuffix\n\t/workspace/pkg/utils/utils.go:55\ngithub.com/Mellanox/network-operator/pkg/state.NewStateOFED\n\t/workspace/pkg/state/state_ofed.go:42\ngithub.com/Mellanox/network-operator/pkg/state.newNicClusterPolicyStates\n\t/workspace/pkg/state/factory.go:73\ngithub.com/Mellanox/network-operator/pkg/state.newStates\n\t/workspace/pkg/state/factory.go:59\ngithub.com/Mellanox/network-operator/pkg/state.NewManager\n\t/workspace/pkg/state/factory.go:34\ngithub.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager\n\t/workspace/controllers/nicclusterpolicy_controller.go:232\nmain.main\n\t/workspace/main.go:87\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371\nfailed to get files from manifest dir\ngithub.com/Mellanox/network-operator/pkg/state.NewStateOFED\n\t/workspace/pkg/state/state_ofed.go:44\ngithub.com/Mellanox/network-operator/pkg/state.newNicClusterPolicyStates\n\t/workspace/pkg/state/factory.go:73\ngithub.com/Mellanox/network-operator/pkg/state.newStates\n\t/workspace/pkg/state/factory.go:59\ngithub.com/Mellanox/network-operator/pkg/state.NewManager\n\t/workspace/pkg/state/factory.go:34\ngithub.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager\n\t/workspace/controllers/nicclusterpolicy_controller.go:232\nmain.main\n\t/workspace/main.go:87\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371\nfailed to create OFED driver State\ngithub.com/Mellanox/network-operator/pkg/state.newNicClusterPolicyStates\n\t/workspace/pkg/state/factory.go:76\ngithub.com/Mellanox/network-operator/pkg/state.newStates\n\t/workspace/pkg/state/factory.go:59\ngithub.com/Mellanox/network-operator/pkg/state.NewManager\n\t/workspace/pkg/state/factory.go:34\ngithub.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager\n\t/workspace/controllers/nicclusterpolicy_controller.go:232\nmain.main\n\t/workspace/main.go:87\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371\nfailed to create state manager\ngithub.com/Mellanox/network-operator/pkg/state.NewManager\n\t/workspace/pkg/state/factory.go:36\ngithub.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager\n\t/workspace/controllers/nicclusterpolicy_controller.go:232\nmain.main\n\t/workspace/main.go:87\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"}
github.com/go-logr/zapr.(*zapLogger).Info
        /go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:126
github.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager
        /workspace/controllers/nicclusterpolicy_controller.go:235
main.main
        /workspace/main.go:87
runtime.main
        /usr/local/go/src/runtime/proc.go:225
panic: Failed to create State manager

goroutine 1 [running]:
github.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager(0xc00001c940, 0x1988148, 0xc0001fcfc0, 0x1977f60, 0xc00059f640)
        /workspace/controllers/nicclusterpolicy_controller.go:236 +0x5f5
main.main()
        /workspace/main.go:87 +0x569
`

sjug commented:
> need to change 'command' param in network-operator/templates/operator.yaml
> 
> This PR doesn't work for me if I install operator using helm even I fixed template

Indeed, I would be surprised if it did work. I didn't change anything regarding helm.

e0ne commented:
> 'make lint' fails in several files

CI should check on the next PR update. I'm not sure if we need to rebase it or not.

sjug commented:
@e0ne The `lint` checks are failing on artifacts that already exist in the repo right now...

adrianchiris commented on Makefile:
```diff
@@ -1,59 +1,61 @@
-TESTPKGS = $(shell env GOPATH=$(GOPATH) $(GO) list -f '{{ if or .TestGoFiles .XTestGoFiles }}{{ .ImportPath }}{{ end }}' $(PKGS))
-
-export GOPATH
-export GOBIN
```
these two would need to be exported in the modified Makefile as well.

also GO<PATH|BIN> initialization logic is different in the modified Makefile.

sjug replied:
I used the new initialization logic, why would we want to preserve the old?

adrianchiris commented on Makefile:
```diff
@@ -1,59 +1,61 @@
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
```
L#51 defines GOPATH if not specified. however here we take the default value.

sjug replied:
I'm not sure what you're trying to say?
`# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)`

adrianchiris replied:
So, ill address both comments here.

to my understanding, previously GOPATH and GOBIN, unless specifically passed to Makefile would take the value `$(CURDIR)/.gopath`, `$(CURDIR)/bin` respectively these would then be exported to reflect the new values as env variables to be used by go.

this was done to avoid modifying the content of your original `GOPATH` and `GOBIN` paths (i think that was the original intention) when you run make from your dev machine / laptop.


now,
GOPATH is set within makefile (L51) but never exported so `go env GOPATH` and `$GOPATH` might have different values.
GOBIN would either have the `$GOBIN` env value (if set) or `$GOPATH/bin` where `GOPATH` is the current env varable value (it will never take the value for `$(CURDIR)/bin`).


its a change in behaviour. also i see CI faling in finding some targets, may be related ?

sjug replied:
I don't see any reason why one would want to "not modify the content of your original `GOPATH` and `GOBIN`. That's the expected and current behavior for any go build. The environment variables can be locally overridden at any time with other paths. 

Master doesn't pass CI (lint) either and yet here we are.

e0ne commented:
Operator doesn't work for me when I deployed it into Kubernetes:
`2021-07-21T07:17:36.766Z        ERROR   controllers.NicClusterPolicy    Error creating state manager.   {"error:": "failed to create state manager: failed to create OFED driver State: failed to get files from manifest dir: error traversing directory tree: lstat /etc/manifests/stage-ofed-driver: no such file or directory", "error:Verbose": "lstat /etc/manifests/stage-ofed-driver: no such file or directory\nerror traversing directory tree\ngithub.com/Mellanox/network-operator/pkg/utils.GetFilesWithSuffix\n\t/workspace/pkg/utils/utils.go:55\ngithub.com/Mellanox/network-operator/pkg/state.NewStateOFED\n\t/workspace/pkg/state/state_ofed.go:42\ngithub.com/Mellanox/network-operator/pkg/state.newNicClusterPolicyStates\n\t/workspace/pkg/state/factory.go:73\ngithub.com/Mellanox/network-operator/pkg/state.newStates\n\t/workspace/pkg/state/factory.go:59\ngithub.com/Mellanox/network-operator/pkg/state.NewManager\n\t/workspace/pkg/state/factory.go:34\ngithub.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager\n\t/workspace/controllers/nicclusterpolicy_controller.go:232\nmain.main\n\t/workspace/main.go:87\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371\nfailed to get files from manifest dir\ngithub.com/Mellanox/network-operator/pkg/state.NewStateOFED\n\t/workspace/pkg/state/state_ofed.go:44\ngithub.com/Mellanox/network-operator/pkg/state.newNicClusterPolicyStates\n\t/workspace/pkg/state/factory.go:73\ngithub.com/Mellanox/network-operator/pkg/state.newStates\n\t/workspace/pkg/state/factory.go:59\ngithub.com/Mellanox/network-operator/pkg/state.NewManager\n\t/workspace/pkg/state/factory.go:34\ngithub.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager\n\t/workspace/controllers/nicclusterpolicy_controller.go:232\nmain.main\n\t/workspace/main.go:87\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371\nfailed to create OFED driver State\ngithub.com/Mellanox/network-operator/pkg/state.newNicClusterPolicyStates\n\t/workspace/pkg/state/factory.go:76\ngithub.com/Mellanox/network-operator/pkg/state.newStates\n\t/workspace/pkg/state/factory.go:59\ngithub.com/Mellanox/network-operator/pkg/state.NewManager\n\t/workspace/pkg/state/factory.go:34\ngithub.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager\n\t/workspace/controllers/nicclusterpolicy_controller.go:232\nmain.main\n\t/workspace/main.go:87\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371\nfailed to create state manager\ngithub.com/Mellanox/network-operator/pkg/state.NewManager\n\t/workspace/pkg/state/factory.go:36\ngithub.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).SetupWithManager\n\t/workspace/controllers/nicclusterpolicy_controller.go:232\nmain.main\n\t/workspace/main.go:87\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"}
`

sjug commented:
@e0ne How are you deploying the operator?

e0ne commented:
/retest-all

sjug commented:
@e0ne As per our discussion yesterday, I have cleaned up the `Makefile` and I have updated the GitHub actions to suit.

For some reason there's this new step "Copy RIght Validation"

Here are the logs:
```
Branch 'pull-requests/206/head' set up to track remote branch 'pull-requests/206/head' from 'origin'.
/jenkins/workspace/copy_right_validator-network-operator
/tmp/k8s_13175/network-operator /jenkins/workspace/copy_right_validator-network-operator
validiting config/crd/patches/cainjection_in_hostdevicenetworks.yaml to match "2021 NVIDIA CORPORATION & AFFILIATES".....
    -Failed!

validiting config/crd/patches/webhook_in_hostdevicenetworks.yaml to match "2021 NVIDIA CORPORATION & AFFILIATES".....
    -Failed!

validiting config/rbac/hostdevicenetwork_editor_role.yaml to match "2021 NVIDIA CORPORATION & AFFILIATES".....
    -Failed!

validiting config/rbac/hostdevicenetwork_viewer_role.yaml to match "2021 NVIDIA CORPORATION & AFFILIATES".....
    -Failed!

validiting config/rbac/service_account.yaml to match "2021 NVIDIA CORPORATION & AFFILIATES".....
    -Failed!

validiting config/samples/mellanox.com_v1alpha1_hostdevicenetwork.yaml to match "2021 NVIDIA CORPORATION & AFFILIATES".....
    -Failed!

validiting manifests/stage-ofed-driver/0001_namespace.yaml to match "2021 NVIDIA CORPORATION & AFFILIATES".....
    -Failed!

/jenkins/workspace/copy_right_validator-network-operator
Failed to validate the projects copyrights!
```

These files do not and will not have a copyright. Can you please fix this check to exclude directories for files that will not be copyrighted.

AbdYsn commented:
/retest-copy-right

e0ne commented on .github/workflows/build.yml:
```diff
@@ -8,14 +8,13 @@ jobs:
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.28.3
+          args: --timeout 10m
```
Why timeout for a linter is so big? Can we decrease it to 3-5 minutes?

sjug replied:
Sometimes it takes a long time, I'd rather have a long timeout to begin with.

e0ne commented on .github/workflows/build.yml:
```diff
@@ -35,18 +36,20 @@ jobs:
         uses: actions/setup-go@v2
         with:
-          go-version: '1.15.8'
+          go-version: '1.16.6'
```
We've got 1.15 in our go.mod, that's why I didn't use 1.16.x

sjug replied:
The new `Dockerfile` is using 1.16.

e0ne commented on Makefile:
```diff
@@ -142,78 +109,6 @@ image: test ## Build docker image with the manager.
 docker-push: ## Push docker image with the manager.
 	${IMAGE_BUILDER} push ${IMG}
 
-##@ Tests
```
Why did you remove these targets? IMO, it's good to have them for development

e0ne replied:
This also affects next commit with github actions modifications. Current approach allows us to use exactly the same commands during development and CI processes

sjug replied:
To begin with these lines are largely the reasons used for justification for the `$(BASE)` and `GOPATH` hackery, so I've removed them to make the `Makefile` clean.

Also the targets/aliases in question are almost entirely unnecessary/unused:
- `lint`: `golangci-lint` it's in the name, it's a linter for CI. Your own go development environment will have a linter built in.
- `lint-dockerfile`: The `Dockerfile` is static and doesn't need a linter, but I still added it to the GitHub actions. 
- `helm lint`: Might be something to retain.
- `test-bench`: Alias target for passing an argument to the test command, never seen this used. What does this provide?
- `test-short`: Alias target for passing an argument to the test command, never seen this used. Why would one want to run only a subset of tests? The full `test` target with verbosity and coverage takes ~15s to run.
- `test-verbose`: Alias target for passing an argument to the test command. This is the default I have set for the `test` target.
- `check test tests`: Very curious why one would want to have three test aliases for a single test target.
- `test-xml`: Unused, `GO2XUNIT` not defined in the `Makefile` and the tool is depricated.
- `test-coverage`: Coverage output as part of new `test` target.
- `test-coverage-tools`: Unused, uploads coverage output to Coveralls.io.

e0ne replied:
I agree about test-* targets but I still want to keep *lint targets as is.

e0ne commented on manifests/stage-ofed-driver/0001_namespace.yaml:
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
```
Please, add license header for all new non-empty files

sjug replied:
None of the manifests have license headers. Why we want to add them now?

adrianchiris replied:
actually they do. (some of them) but this is not needed for new files. (internal process that changed..)

however, i see a greater problem.
This namespace needs to be created when we deploy operator. i.e when deploying with:
- make deploy
- helm chart
- olm

stage OFED driver can be skipped in general, moreover this is not the first stage that is applied.

moshe010 commented:
/retest-all

e0ne commented:
/retest-nic_operator_kind

moshe010 commented:
@sjug Thank for this PR. We decided that we will fork the changes on the update SDK and add you as co-author and @e0ne  will continue the Makefile change so we will better understand the SDK update change. The reason is that your PR introduce many changes in the Makefile targets which we don't want to mix in this PR. So we will do Makefile changes just to support the new sdk. later on we will see how to improve the Makefile base on your comments and guidelines from other existing operator. Thanks for you effort spending on this PR and hope to see more contribution in the future :)

sjug commented:
@moshe010 The goal of this PR is to repair the `operator-sdk` fucntionality that was removed from this operator in the past, so that we can use the `operator-sdk` to generate the bundles necessary for OperatorHub.

There's a reason that the `Makefile` changes are included in this PR with the `operator-sdk` update.

The current state of the deployments and the `Makefile` entirely break the `operator-sdk` functionality.

Currently:
```
.PHONY: deploy
deploy: manifests	## Deploy controller in the configured Kubernetes cluster in ~/.kube/config
	@scripts/deploy-operator.sh
```

Correct (PR):
```
deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
	$(KUSTOMIZE) build config/default | kubectl apply -f -
```

So updating the PR without the `Makefile` is largely pointless.

moshe010 commented:
> @moshe010 The goal of this PR is to repair the `operator-sdk` fucntionality that was removed from this operator in the past, so that we can use the `operator-sdk` to generate the bundles necessary for OperatorHub.
> 
> There's a reason that the `Makefile` changes are included in this PR with the `operator-sdk` update.
> 
> The current state of the deployments and the `Makefile` entirely break the `operator-sdk` functionality.
> 
> Currently:
> 
> ```
> .PHONY: deploy
> deploy: manifests	## Deploy controller in the configured Kubernetes cluster in ~/.kube/config
> 	@scripts/deploy-operator.sh
> ```
> 
> Correct (PR):
> 
> ```
> deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
> 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
> 	$(KUSTOMIZE) build config/default | kubectl apply -f -
> ```
> 
> There have already been other standalone `operator-sdk` "update" commits/PRs made to this operator, but they don't actually **fix** this issues caused by the `Makefile`.
> 
> So updating the PR without the `Makefile` is largely pointless.

I didn't say we won't do makefile change, but you change is too big and remove target that at the moment we don't think is necessary to remove. As I mention above we will do the necessary changes to make sdk 1.8 work. even today SR-IOV operator is not using deploy and undeploy as you mention [1] so not sure how critical it is. I do understand that we need to add the bundle-build target for olm. Any others that are needed? 

[1]- https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/Makefile#L102-L109

sjug commented:
@moshe010 The SR-IOV operator is far more simple from a deployment perspective, you should take a look at a more similar NVIDIA operator like the GPU Operator [1].

It's not just the specifics of the `make deploy` target, but rather the fact that `kustomize` will process the manifests inside the `config` directory and then use that rather than some hardcoded files.

My goal like I said is to add OpenShift compatibility and add this operator to OperatorHub. The easiest way to generate the manifests for inclusion in OperatorHub is by using the built in `operator-sdk` targets like `make bundle` [2] (commented out and non-functional in the current `Makefile`). Whether we "need" to use `make deploy` correctly is irrelevant, as it functions like a functional test for other make targets (like `make bundle`) that use `kustomize` and the same method of dynamically generating manifests. 

[1]- https://github.com/NVIDIA/gpu-operator/blob/master/Makefile
[2]- https://sdk.operatorframework.io/docs/olm-integration/quickstart-bundle/#creating-a-bundle

moshe010 commented:
> @moshe010 The SR-IOV operator is far more simple from a deployment perspective, you should take a look at a more similar NVIDIA operator like the GPU Operator [1]. Oddly the GPU Operator `Makefile` is quite similar to the one in my PR and also doesn't have any of the targets that you want to retain.
> 
> It's not just the specifics of the `make deploy` target, but rather the fact that `kustomize` will process the manifests inside the `config` directory and then use that rather than some hardcoded files.
> 
> My goal like I said is to add OpenShift compatibility and add this operator to OperatorHub. The easiest way to generate the manifests for inclusion in OperatorHub is by using the built in `operator-sdk` targets like `make bundle` [2] (commented out and non-functional in the current `Makefile`). Whether we "need" to use `make deploy` correctly is irrelevant, as it functions like a functional test for other make targets (like `make bundle`) that use `kustomize` and the same method of dynamically generating manifests.
> 
> [1]- https://github.com/NVIDIA/gpu-operator/blob/master/Makefile
> [2]- https://sdk.operatorframework.io/docs/olm-integration/quickstart-bundle/#creating-a-bundle

@sjug  we share the same goal, but it seem that you want to fix "everything" and we want to do the minimal changes to support openshift in this PR. 
few questions: 
Why we need to upgrade the SDK version?
Can we just add bundle and bundle-build targets [1] ? 
Can we test it on vanilla k8s? 
[1] - https://github.com/NVIDIA/gpu-operator/blob/master/Makefile#L127-L136

sjug commented:
@moshe010 I don't want to fix "everything," I just want to have a functional PR and operator. This `Makefile` is a mess and yet you are concerned about preserving it. 

>  Why we need to upgrade the SDK version?
The bundle generation continues to evolve, so we want a modern version that supports what we're trying to do.

>  Can we just add bundle and bundle-build targets [1] ? It uses heavy hacks in 
The hackery involved in this `Makefile` is unique as I have never seen another like it before. Every target is wrapped in order to create new $GOPATH within the subdirectory and work within it. There's no reason for it to function this way other than excessive complexity, incompatibility and fragility that is introduced.
Yes, it may be possible to just add the two targets. If you or someone on your team can contribute a `Makefile` that works with the rest of the PR I'd be more than happy to take it. There's no reason to keep a lot of the targets, such as `make deploy`, there's no benefit to maintaining **yet another** set of manifests. It's just more duplication...

> Can we test it on vanilla k8s?
Yes you can, I have not tested it on vanilla k8s and I've been asking for this testing at our weekly meetings. So it would be great if you folks would do this testing.

moshe010 commented:
will support opendhift support sdk1.3? what is the minimal sdk version the openshift support?

adrianchiris commented on scripts/check-manifests.sh:
```diff
@@ -1,24 +0,0 @@
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-# This script has been designed to run on CI only so it supposes that
```
IIUC, this is intended to catch if changes were made that affect CRDs but were (accidently) omitted from the PR.
from the comment, this bit runs in CI.

so either:
- this is indeed no longer needed, and we should remove the check from our CI (mellanox-ci)
- this is needed and was accidently removed

adrianchiris replied:
OK Checked this with @e0ne, this was used in our old travis CI, when moving to github action the check was accedently removed and he should add it back. 

please dont remove this file in this PR.

sjug commented:
Closing, moving forward with https://github.com/Mellanox/network-operator/pull/233
